### PR TITLE
calendar event bug fixes

### LINKED ISSUE #1045: Calendar event end date is buggy.
Currently, when creating events, end date gets a reset when submitted and we see end date equivalent to start date.
Screenshot 2023-10-16 at 2 49 34 PM

### PR DESCRIPTION
**Pull request by gtalha07**
- fixes #1045.
- Updated Riverpod to latest version (2.4.4).
- Remove pending status from rsvp selection. (Events will still show pending status if not responded yet).
- Merge start/end time and date selection textfields.
- Remove link input textfields.

### CODE REVIEW

File: app/lib/features/events/sheets/edit_event_sheet.dart [hunk 1]
```diff
@@ -165,41 +164,6 @@ class _EditEventSheetConsumerState extends ConsumerState<EditEventSheet> {
...
-                            ),
-                          ),
-                        ),
-                      ),
-                    ],
-                  ),
-                ),
               ],
             ),
             const SizedBox(height: 15),
```
Comment (gnunicorn) [inline app/lib/features/events/sheets/edit_event_sheet.dart:169]: so we have a SizedBox above a SizedBox here now? That seems odd...

**Review by gnunicorn — APPROVED** [review]

File: app/lib/features/events/sheets/edit_event_sheet.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/events/sheets/edit_event_sheet.dart:169]: Ahh might have been missed along with other removal of code-blocks....

File: app/lib/features/events/pages/event_page.dart [hunk 2]
```diff
@@ -59,7 +59,7 @@ class CalendarEventPage extends ConsumerWidget {
                 title: 'Remove this post',
                 eventId: event.eventId().toString(),
                 onSuccess: () {
-                  ref.invalidate(calendarEventProvider);
+                  ref.invalidate(calendarEventProvider(calendarId));
```
Comment (bitfriend) [inline app/lib/features/events/pages/event_page.dart:62]: `calendarId` is needed?

File: app/lib/features/events/pages/event_page.dart [hunk 3]
```diff
@@ -325,14 +325,11 @@ class CalendarEventPage extends ConsumerWidget {
...
                           rsvp.first.name,
+                          ref,
                         );
-                        // refresh rsvp state and event
-                        myRsvpStatus =
-                            ref.refresh(myRsvpStatusProvider(calendarId));
-                        event = ref.refresh(calendarEventProvider(calendarId));
-                        EasyLoading.dismiss();
+                        ref.invalidate(myRsvpStatusProvider);
+                        ref.invalidate(calendarEventProvider);
```
Comment (bitfriend) [inline app/lib/features/events/pages/event_page.dart:332]: The argument `calendarId` is not needed?

File: app/lib/features/events/pages/event_page.dart [hunk 4]
```diff
@@ -372,22 +369,16 @@ class CalendarEventPage extends ConsumerWidget {
...
   ) async {
     EasyLoading.show(status: 'Updating RSVP', dismissOnTap: false);
-    if (event == null) {
-      EasyLoading.showError(
-        'Error fetching event details',
-        duration: const Duration(seconds: 2),
-      );
-      return;
-    }
+    final event = await ref.read(calendarEventProvider(calendarId).future);
```
Comment (bitfriend) [inline app/lib/features/events/pages/event_page.dart:376]: It is nice work that you removed `CalendarEvent? event` from function argument.

File: app/lib/features/events/pages/event_page.dart (hunk 2 above)
Comment (gtalha07) [reply app/lib/features/events/pages/event_page.dart:62]: Yeah its not necessary

**Pull request merged**